### PR TITLE
Implementing version regex in translations

### LINKED ISSUE #23518: Vulnerability scanner - Missing functionality on translations
## Description
Currently, the translation mechanism used on the vulnerability scanner (https://github.com/wazuh/wazuh/issues/22477):

- Doesn't validate that the vendor name matches the one on the translation regex
- Lacks the capability of replacing the version provided by syscollector

### Static version
Replace the version with some fixed value:
```json
{
  "target": [
    "windows"
  ],
  "source": {
    "vendor": "^Microsoft Corporation",
    "product": "^Microsoft Office Professional Plus 2016 - en-us$",
    "version": ""
  },
  "translation": [
    {
      "vendor": "microsoft",
      "product": "office",
      "version": "16.0.0.0"
    }
  ]
}
```

### Dynamic version
Obtain the version information from the package name:
```json
  {
    "scan_id": 0,
    "scan_time": "2024/05/14 14:13:18",
    "format": "win",
    "name": "Java 2 Runtime Environment Standard Edition v1.3.1", <----- VERSION IN NAME
    "version": " ", <----- EMPTY VERSION
    "priority": " ",
    "section": " ",
    "size": 0,
    "vendor": " ",
    "install_time": "2024/05/14 14:12:38",
    "architecture": "i686",
    "multiarch": null,
    "source": " ",
    "description": " ",
    "location": " ",
    "cpe": null,
    "msu_name": null,
    "checksum": "07d6fe0244cbe8abad0cdc83ae6d1f53dcd092da",
    "item_id": "8d91907b799e329bf1c472fe5898abfa946ee746"
  }
```

In this example, we could use the following translation
```json
{
  "target": [
    "windows"
  ],
  "source": {
    "vendor": "^\\s",
    "product": "^Java \\d Runtime Environment Standard Edition v\\d+\\.\\d+\\.\\d+",
    "version": "v(\\d+\\.\\d+\\.\\d+)"
  },
  "translation": [
    {
      "vendor": "oracle",
      "product": "jre",
      "version": ""
    },
    {
      "vendor": "sun",
      "product": "jre",
      "version": ""
    }
  ]
}
```

## Logic
```mermaid
flowchart TD
    A[Start] --> B{source.version empty}
    B -->|No| C[Use version extracted from product name]
    B -->|Yes| E{translation.version empty}
    E -->|No| F[Use translation version]
    E -->|Yes| G[No version to replace]
```

## DoD
- [x] Functionality implemented
- [x] Tests updated

### PR DESCRIPTION
**Pull request by pereyra-m**
|Related issue|
|---|
|Closes #23518|

## Description

This PR adds the vendor and version regex validations when a package is scanned.

- UTs and QA tests were added.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation
- [x] QA templates contemplate the added capabilities
- [x] Added unit tests (for new features)

### CODE REVIEW

**Review by sebasfalcone — COMMENTED** [review]
Great work @pereyra-m!


Partial review:
- Add the documentation for test-case 18

Testing pending

**Review by sebasfalcone — CHANGES_REQUESTED** [review]
Lets add the documentation for the 018 test

File: src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp [hunk 1]
```diff
@@ -462,13 +464,21 @@ class TDatabaseFeedManager final : public Observer<nlohmann::json&>
...
+                                   : std::nullopt;
+            auto versionRegex = queryData->source()->version() && !queryData->source()->version()->str().empty()
+                                    ? std::optional<std::regex>(queryData->source()->version()->str())
+                                    : std::nullopt;
 
             // Initialize Translation object to store translation data
             Translation translationQuery = {};
             translationQuery.productRegex = std::move(productRegex);
             translationQuery.vendorRegex = std::move(vendorRegex);
+            translationQuery.versionRegex = std::move(versionRegex);
```
[the remark this answers is not in the document]
Comment (pereyra-m) [reply src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp]: Thank you!
It's more readable now.

**Review by sebasfalcone — CHANGES_REQUESTED** [review]
Rebase needed

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**